### DISCUSSION TITLE
Updating simulator

### DISCUSSION THREAD

**Discussion by infnorm** [grblHAL/core#369] [General]
@terjeio, not sure this warrants a discussion. I've been using the simulator to develop my improvements to bCNC. After a long period of no time, I'm finally getting back to it. 

- I did a pass to make the simulator compile and work again based on HAL changes.
- It consists of these pull requests
  * #367 #368 
  * https://github.com/grblHAL/Simulator/pull/6/commits
- I'm not terribly familiar with the grblHAL code, so I'm happy to fix any mistakes I made.

Do we have any sort of automated continuous integration. It might make sense to try to keep the simulator working as a a smoke test for any changes? I'm happy to help in this regard.

**Comment by terjeio** [comment]
> Do we have any sort of automated continuous integration. 

Not really, it is set up for the ESP32 driver by someone else but it generally fails. To make automated tests meaningful it has to run with a selection of option/plugin combinations? An the the simulator falls short since it currently cannot be compiled with plugins?

If you have ideas I am all ears.
